Update to python 3

JamesWrigley opened the pull request:
Hi there,
I got some errors when I tried to use pyrc with a python 3 project, so I made a few changes for it to work with python 3. Though it probably breaks backward compatibility with python 2, so I dunno if you'd want to merge it.

Anyway, no pressure, just thought I'd put it out there :)

JamesWrigley committed:
Started updating pyrc to work with python 3

JamesWrigley committed:
Updated pyrc to work with python 3. Not fully tested yet!

JamesWrigley committed:
Minor fix to allow joining of multiple channels

dmtucker commented on pyrc/bots.py:
```diff
@@ -131,7 +132,7 @@ def parsefuncs(self, target, sender, message, funcs):
         if group_dict and (len(groups) > len(group_dict)):
           # match.groups() also returns named parameters
-          raise "You cannot use both named and unnamed parameters"
+          raise IOError("You cannot use both named and unnamed parameters")
```
Maybe `TypeError` or `ValueError` would be better here?

dmtucker commented on pyrc/bots.py:
```diff
@@ -95,8 +96,8 @@ def run_listeners(self, line):
   def addhooks(self):
-    for func in self.__class__.__dict__.values():
-      if callable(func) and hasattr(func, '_type'):
+    for func in list(self.__class__.__dict__.values()):
```
I think `for func in vars(self.__class__).values():` would work.

dmtucker commented on pyrc/bots.py:
```diff
@@ -5,7 +5,8 @@
 
-import threads
+from . import threads
+import collections
```
nit: PEP8 would put `collections` in the group above.

JamesWrigley committed:
Code cleanup

Reorder imports, use vars(), and throw a TypeError instead of an IOError when
the user passes in named and unnamed parameters.

JamesWrigley commented:
One sec, fixing the conflicts now.

JamesWrigley committed:
Merge branch 'master' of https://github.com/sarenji/pyrc

Note: now we use func.__name__ instead of func.func_name in hooks.py.
